src/main/java/requestMgmt/RequestManager.java

```diff
@@ -0,0 +1,323 @@
+package requestMgmt;
+
+import com.google.common.collect.BiMap;
+import dataMgmt.DataManager;
+import dataMgmt.SessionStorage;
+import websocket.IRequestSendErrorHandler;
+import websocket.WSManager;
+import websocket.models.File;
+import websocket.models.Permission;
+import websocket.models.Project;
+import websocket.models.Request;
+import websocket.models.requests.*;
+import websocket.models.responses.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by fahslaj on 10/15/2016.
+ */
+public abstract class RequestManager {
+
+    private DataManager dataManager;
+    private WSManager wsManager;
+
+    private IRequestSendErrorHandler requestSendErrorHandler;
+    private IInvalidResponseHandler invalidResponseHandler;
+
+    public RequestManager(DataManager dataManager, WSManager wsManager,
+                          IRequestSendErrorHandler requestSendErrorHandler,
+                          IInvalidResponseHandler invalidResponseHandler) {
+        this.dataManager = dataManager;
+        this.wsManager = wsManager;
+        this.requestSendErrorHandler = requestSendErrorHandler;
+        this.invalidResponseHandler = invalidResponseHandler;
+    }
+
+    /**
+     * Login to the CodeCollaborate server.
+     * @param username username to log in with
+     * @param password password for the given username
+     */
+    public void login(String username, String password) {
+        Request loginRequest = new UserLoginRequest(username, password).getRequest(response -> {
+            UserLoginResponse loginResponse = (UserLoginResponse) response.getData();
+            int status = response.getStatus();
+            if (status == 200) {
+                SessionStorage storage = this.dataManager.getSessionStorage();
+                storage.setUsername(username);
+                storage.setAuthenticationToken(loginResponse.getToken());
+                this.wsManager.setAuthInfo(username, loginResponse.getToken());
+            } else {
+                this.invalidResponseHandler.handleInvalidResponse(status, "Could not login to the CodeCollaborate server.");
```

Comment:
hmm, is invalid the right word for this name? it seems more like it's an incorrect response or unexpected response. Invalid makes it sound like the response object itself was messed up.